thirdparty/onnx/CMakeLists.txt

```diff
@@ -44,20 +44,27 @@ foreach(_onnx_target onnx onnx_proto)
         $<TARGET_PROPERTY:protobuf::libprotobuf,INTERFACE_INCLUDE_DIRECTORIES>)
 endforeach()
 
-if(WIN32)
-    # from onnx==1.13.1 it requires C++17 when compiling on Windows
-    target_compile_features(onnx PRIVATE cxx_std_17)
+if(MINGW)
     # OPTIONAL is a reserved word for mingw at least
-    if(MINGW)
-        target_compile_definitions(onnx PRIVATE OPTIONAL=OPTIONAL_PLACEHOLDER)
-    endif()
+    target_compile_definitions(onnx PRIVATE OPTIONAL=OPTIONAL_PLACEHOLDER)
 endif()
 
-# WA to support old compiler version, removes unused header which may block build
+# from onnx==1.13.1 it requires C++17 when compiling on Windows, and since onnx==1.16.0 on Linux
+target_compile_features(onnx PRIVATE cxx_std_17)
+
+# WA to allow build on Debian 9
 if(CMAKE_COMPILER_IS_GNUCXX AND LINUX AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 7.0)
-    file(READ onnx/onnx/defs/parser.cc CONTENT)
-    string(REPLACE "#include <string_view>" "" CONTENT "${CONTENT}")
-    file(WRITE onnx/onnx/defs/parser.cc "${CONTENT}")
+    file(READ onnx/onnx/common/file_utils.h CONTENT)
+    string(REPLACE "#include <filesystem>" "" CONTENT "${CONTENT}")
+    string(REPLACE "std::filesystem::path proto_u8_path = std::filesystem::u8path(proto_path);" "" CONTENT "${CONTENT}")
+    string(REPLACE "proto_u8_path" "proto_path" CONTENT "${CONTENT}")
+    file(WRITE onnx/onnx/common/file_utils.h "${CONTENT}")
+
+    file(READ onnx/onnx/defs/traditionalml/utils.h CONTENT)
+    string(REPLACE "const auto& [type, length]" "const std::pair<int, int>& type_length" CONTENT "${CONTENT}")
+    string(REPLACE "type !=" "type_length.first !=" CONTENT "${CONTENT}")
+    string(REPLACE "length !=" "type_length.second !=" CONTENT "${CONTENT}")
+    file(WRITE onnx/onnx/defs/traditionalml/utils.h "${CONTENT}")
```

Comment:
I think we can drop these WAs as we migrated to newer Debian ARM version.

Reply:
Agree, let's try

Reply:
Removed